src/java.base/share/classes/jdk/internal/foreign/SystemLookup.java

```diff
@@ -87,10 +87,23 @@ private static SymbolLookup makeAixLookup() {
         return name -> {
             Objects.requireNonNull(name);
             try {
-                long addr = lib.lookup(name);
-                return (addr == 0) ?
-                        Optional.empty() :
-                        Optional.of(MemorySegment.ofAddress(MemoryAddress.ofLong(addr), 0, MemorySession.global()));
+                MemoryAddress funcAddr = null;
+                AixFuncSymbols symbol = AixFuncSymbols.valueOfOrNull(name);
+                if (symbol == null) {
+                    /* Look up the libc functions in the default library. */
+                    funcAddr = MemoryAddress.ofLong(lib.lookup(name));
+                } else {
+                    /* Directly load the specified library with the libc functions
+                     * missing in the default library.
+                     */
+                    SymbolLookup funcsLibLookup =
+                            libLookup(libs -> libs.load(jdkLibraryPath("syslookup")));
+                    MemorySegment funcs = MemorySegment.ofAddress(funcsLibLookup.lookup("funcs").orElseThrow().address(),
+                        ADDRESS.byteSize() * AixFuncSymbols.values().length, MemorySession.global());
+                    funcAddr = funcs.getAtIndex(ADDRESS, symbol.ordinal());
+                }
+
+                return Optional.of(MemorySegment.ofAddress(funcAddr, 0L, MemorySession.global()));
             } catch (NoSuchMethodException e) {
```

Comment:
Can this be moved to a smaller scope?

Reply:
Updated against the suggestion as the exception is only thrown from `lib.lookup(name)`.